### PR TITLE
[DATA-3497] API Improvements

### PR DESCRIPTION
**Pull request by dvfeinblum**
Adds headers, bypasskey, and host to the lib.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -142,7 +142,7 @@
                     <dependency>
                         <groupId>com.puppycrawl.tools</groupId>
                         <artifactId>checkstyle</artifactId>
-                        <version>8.26</version>
+                        <version>8.29</version>
```
Comment (dvfeinblum) [inline pom.xml]: Per the bot.

File: src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java [hunk 2]
```diff
@@ -63,6 +67,21 @@ public static JWPlatformClient create(final String apiKey, final String apiSecre
...
   }
 
+  /**
+   * Instantiate a new {@code JWPlatformClient} instance.
+   *
+   * @param apiSecret - your api key
+   * @param apiKey - your api secret
+   * @return a {@code JWPlatformClient} instance
+   */
+  public static JWPlatformClient create(final String apiKey, final String apiSecret, final String host) {
```
Comment (swhelan091) [inline src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java]: Layer the other create().

```
public static JWPlatformClient create(final String apiKey, final String apiSecret) {
   return create(apiKey, apiSecret ,"https://api.jwplatform.com/v1/");
}
```

Then you can get rid of the constructor: `private JWPlatformClient(final String apiKey, final String apiSecret)`

File: src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java [hunk 3]
```diff
@@ -233,20 +260,22 @@ public JSONObject request(final String path, final String requestType)
...
         case "GET":
           requestUrl = this.buildRequestUrl(host, path, params);
-          response = Unirest.get(requestUrl).asJson();
+          response = Unirest.get(requestUrl).headers(headers).asJson();
           break;
         case "POST":
           if (isBodyParams) {
             requestUrl = this.buildRequestUrl(host, path, Collections.emptyMap());
-            response = Unirest.post(requestUrl).body(new JSONObject(params)).asJson();
+            System.out.println(Unirest.post(requestUrl).toString());
```
Comment (swhelan091) [inline src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java]: delete

File: src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java [hunk 4]
```diff
@@ -188,26 +207,34 @@ private JSONObject uploadVideo(final String uploadPath, final String localFilePa
...
-          throws JWPlatformException {
-    return this.request(path, new HashMap<>(), false, requestType);
+      throws JWPlatformException {
+    return this.request(path, new HashMap<>(), false, requestType, new HashMap<>());
+  }
+
+  /**
+   * see {@link #request(String, Map, boolean, String, Map)}.
+   */
+  public JSONObject request(final String path, final Map<String, String> params, final boolean isBodyParams, final String requestType)
```
Comment (swhelan091) [inline src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java]: Don't update this one. It'd be a breaking change. If someone was using this request() function, it'd now be gone. Overload one more with the added header param.

File: src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java [hunk 5]
```diff
@@ -57,10 +57,22 @@ private JWPlatformClient(final String apiKey, final String apiSecret) {
...
    */
   public static JWPlatformClient create(final String apiKey, final String apiSecret) {
+    return create(apiKey, apiSecret, "https://api.jwplatform.com/v1/");
+  }
+
+  /**
+   * Instantiate a new {@code JWPlatformClient} instance.
+   *
+   * @param apiSecret - your api key
+   * @param apiKey - your api secret
```
Comment (swhelan091) [inline src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java]: missing doc string for host

File: src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java [hunk 6]
```diff
@@ -167,11 +179,12 @@ private String buildRequestUrl(
    * @return - JSON response from JW Platform API
    * @throws JWPlatformException - API returned an exception
    */
-  private JSONObject uploadVideo(final String uploadPath, final String localFilePath)
+  private JSONObject uploadVideo(final String uploadPath, final String localFilePath, final Map<String, String> headers)
```
Comment (swhelan091) [inline src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java]: missing doctring for headers

File: src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java [hunk 7]
```diff
@@ -233,20 +254,21 @@ public JSONObject request(final String path, final String requestType)
    *         {@code e.getCause().getMessage()}
    */
   public JSONObject request(final String path, final Map<String, String> params,
```
Comment (swhelan091) [inline src/main/java/com/jwplayer/jwplatform/JWPlatformClient.java]: missing docstring for headers

**Pull request merged**